### PR TITLE
Pass title through on create and allow editing

### PR DESCRIPTION
**Pull request by clloyd**
screen shot 2017-02-27 at 15 25 49

Pass through title on create and allow editing on the edit screen

### CODE REVIEW

File: public/js/components/AtomEdit/AtomEdit.js [hunk 1]
```diff
@@ -36,11 +31,7 @@ class AtomEdit extends React.Component {
...
 
-  render () {
-
-    if (!this.props.atom) {
-      return <div>Loading...</div>;
-    }
+  renderSpecificEditor () {
 
     //TODO: This is brittle, can we improve?
     const atomType = this.props.atom.atomType.toLowerCase();
```
Comment (shaundillon) [inline public/js/components/AtomEdit/AtomEdit.js]: I think we can get this from the key inside `data` now. Not great but at least it's the same case

File: public/js/components/AtomEdit/AtomEditHeader.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        </div>
+        <div className="atom-card__details">
+          <h3 className="atom-card__heading">{this.props.atom.atomType} Atom</h3>
+          <ManagedForm data={this.props.atom} updateData={this.props.onUpdate}>
+            <ManagedField fieldLocation="title" name="Title:">
+              <FormFieldTextInput/>
+            </ManagedField>
+          </ManagedForm>
+          <div className="atom-card__description">
+            <div><b>Created: </b>{this.renderDate('created')}</div>
```
Comment (shaundillon) [inline public/js/components/AtomEdit/AtomEditHeader.js]: I think this should be a `<ul>`

File: public/js/components/AtomEdit/AtomEdit.js (hunk 1 above)
Comment (clloyd) [inline public/js/components/AtomEdit/AtomEdit.js]: not sure I know what you mean?

File: public/js/components/AtomEdit/CustomEditors/RecipeEditor.js [hunk 3]
```diff
@@ -24,35 +24,32 @@ export class RecipeEditor extends React.Component {
...
-            </FormFieldArrayWrapper>
-          </ManagedField>
-          <ManagedField fieldLocation="data.recipe.images" name="Images">
-            <FormFieldArrayWrapper>
-              <FormFieldImageSelect gridUrl={this.props.config.gridUrl}/>
-            </FormFieldArrayWrapper>
-          </ManagedField>
-        </ManagedForm>
-      </div>
+      <ManagedForm data={this.props.atom} updateData={this.props.onUpdate}>
```
Comment (shaundillon) [inline public/js/components/AtomEdit/CustomEditors/RecipeEditor.js]: Really like that these are just a managed form now.

File: public/js/services/AtomsApi.js [hunk 4]
```diff
@@ -4,44 +4,46 @@ export default {
 
   getAtom: (atomType, atomId) => {
     return pandaFetch(
-      `/api/preview/${atomType}/${atomId}`,
-      {
-        method: 'get',
-        credentials: 'same-origin'
-      }
+      new Request(
```
Comment (shaundillon) [inline public/js/services/AtomsApi.js]: These didn't seem to work when I changed polyfills before, so I was just sending them separately. `pandaFetch.js` may need it's params changing.

File: public/js/components/AtomEdit/AtomEdit.js (hunk 1 above)
Comment (shaundillon) [inline public/js/components/AtomEdit/AtomEdit.js]: the object immediately inside `data` has the same key as the type of the atom. Not the nicest but it'll get around the use of `toLowerCase`

**Pull request merged**